apps/backend/src/schemas/proxy_config.py

```
@@ -176,18 +184,22 @@ class ProxyConfigValidation(BaseModel):
-    """Tracked change in proxy configuration"""
+    """Tracked change in proxy configuration - integrated with ConfigurationSnapshot"""
 
     model_config = ConfigDict(from_attributes=True)
 
     config_id: int = Field(..., description="Configuration ID")
+    snapshot_id: UUID = Field(..., description="Configuration snapshot ID")
+    change_event_id: Optional[UUID] = Field(None, description="Configuration change event ID")
     change_type: str = Field(..., description="Type of change (created, modified, deleted)")
     old_hash: Optional[str] = Field(None, description="Previous file hash")
     new_hash: Optional[str] = Field(None, description="New file hash")
     changes_detected: List[str] = Field(
```

Ruff (0.12.2)

Use `list` instead of `List` for type annotation

Replace with `list`

(UP006)